pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -221,14 +275,17 @@ func TestCreatePod(t *testing.T) {
 		fakeClientset := fake.NewSimpleClientset()
 		controller := NewNoExecuteTaintManager(ctx, fakeClientset, (&podHolder{pod: item.pod}).getPod, getNodeFromClientset(fakeClientset), getPodsAssignedToNode(fakeClientset))
 		controller.recorder = testutil.NewFakeRecorder()
-		go controller.Run(ctx)
 		controller.taintedNodes = item.taintedNodes
+		go controller.Run(ctx)
 		controller.PodUpdated(nil, item.pod)
-		// wait a bit
-		time.Sleep(timeForControllerToProgress)
+
+		actions := fakeClientset.Actions()
+		if item.expectDelete {
```

Comment:
Instead of the if, pass if the delete is expected (maybe actually pass the set of pods for which the delete is expected)